### ISSUE TITLE
Inference error for finetuned models.

### ISSUE THREAD

**Issue by Danielement321** [DAMO-NLP-SG/VideoLLaMA2#126]
Thanks for your excellent work!

I've fine-tuned the audio-visual model with custom data in the given format. However, when I used the tuned version, an error occurred like the attached photo. I've checked the `images` for official tuned model from huggingface and my tuned model, and found that the `images` between the two versions are different. Could you please check the issue?

Thanks again for your outstanding work!

```
Loading additional VideoLLaMA weights...
Loading LoRA weights...
Merging LoRA weights...
Model is loaded...
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
IMPORTANT: You are using gradio version 3.50.0, however version 4.44.1 is available, please upgrade.
--------
Traceback (most recent call last):
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/gradio/routes.py", line 534, in predict
    output = await route_utils.call_process_api(
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/gradio/route_utils.py", line 226, in call_process_api
    output = await app.get_blocks().process_api(
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/gradio/blocks.py", line 1550, in process_api
    result = await self.call_function(
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/gradio/blocks.py", line 1185, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 2441, in run_sync_in_worker_thread
    return await future
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 943, in run
    result = context.run(func, *args)
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/gradio/utils.py", line 661, in wrapper
    response = f(*args, **kwargs)
  File "/root/vl/videollama2/serve/gradio_web_server_adhoc_av.py", line 187, in generate
    text_en_out = handler.generate(data, message, temperature=temperature, top_p=top_p, max_output_tokens=max_output_tokens)
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/root/vl/videollama2/serve/gradio_web_server_adhoc_av.py", line 88, in generate
    response = mm_infer(tensor, message, self.model, self.tokenizer, modal=modal.strip('<>'),
  File "/root/vl/./videollama2/__init__.py", line 104, in mm_infer
    output_ids = model.generate(
  File "/root/anaconda3/envs/vl/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/root/vl/./videollama2/model/videollama2_mistral.py", line 131, in generate
    ) = self.prepare_inputs_labels_for_multimodal(
  File "/root/vl/./videollama2/model/videollama2_arch.py", line 217, in prepare_inputs_labels_for_multimodal
    mm_features = self.encode_images_or_videos(images)
  File "/root/vl/./videollama2/model/videollama2_arch.py", line 170, in encode_images_or_videos
    data_batch = torch.stack(data_batch, dim=0)
TypeError: expected Tensor as element 0 in argument 0, but got dict
```

**Comment by Danielement321** [comment]
The problem is caused by the missing audio tower and I have solved this by adding the path in my config file.

**Comment by ffcarina** [comment]
@Danielement321 Hi! Do you train your own audio-visual model on the basis of VideoLLaMA2.1-7B-16F or continue to fine-tune VideoLLaMA2.1-7B-AV?

**Comment by Danielement321** [comment]
Hey there!
I tried to continue to fine-tune `VideoLLaMA2.1-7B-AV` with lora. However, I've met more problems than the one mentioned in this issue. I loaded the model by `--model_path models/DAMO-NLP-SG/VideoLLaMA2.1-7B-AV` and `--lora_enable True --lora_r 128 --lora_alpha 256`, but even when set `LOCAL_BATCH_SIZE=1` for each GPU, there is still OOM error. By setting `--tune_adapter_llm False`, the script can run.
Besides, there might be some problems in this script, because I set the learning rate to be 0 and train very few samples, and found the 'trained' model output a lot messy codes.
